### ISSUE TITLE
Use of python_requires with enabled short_paths fails

### ISSUE THREAD

**Issue by kdsx** [conan-io/conan#5814]
In Windows conan fails when a recipe uses `python_requires` and `short_paths = True` at the same time.

To reproduce the issue, it's enough to just add `short_paths = True` into the consumer conanfile of the `python_requires` example.

Conan will end with
```
ERROR: Error while trying to get recipe sources for pyreq/version@user/channel. No remote defined
```

**Comment by czoido** [comment]
Hi @kdsx,
I was able to reproduce the issue when using `export_sources` in the `python_requires`
While we have a look at it, did you try to set the `short_paths = True` in the python requires as well?

Thanks a lot for the feedback

**Comment by kdsx** [comment]
Hi @czoido,
I've already tried it. Unfortunately with no effect. Conan fails with the same message.

**Comment by memsharded** [comment]
Hi!

I haven't been able to reproduce it, please @czoido share a unit test that can reproduce it.

That error message typically appears when a package is downloaded from a remote, then the remote is removed (and the necessary sources aren't there), so when the package need the sources it fails. It is not that directly related to short_paths, but to the remote removal. So any help with more detailed steps to reproduce will help. Thanks!

**Comment by kdsx** [comment]
In my case it happened for a package which exists only in the local cache and has never been to a remote server. Then for simplicity I've reproduced it on the mentioned conan example. In contrast to the example my package does not use `exports_sources`, only `exports` which appear in the local cache in the same directory as `conanfile.py`, thus it should not require any additional sources to download.

I use the latest conan version at the moment (1.18.5) on Windows 10. I doubt it's important, but I run conan from cygwin environment, however conan itself installed by `pip` into `venv` generated by a native (non-cygwin) version of Python 3.

**Comment by kdsx** [comment]
I've looked into code a bit and found that when conan installs the dependencies (including pyreq) it calls `_build_package` and then `complete_recipe_sources` for every dependency and the package itself. In the very beginning it tries to find the sources folder and then retrieve it if it is not available. The problem is probably here:
```
sources_folder = cache.package_layout(ref, conanfile.short_paths).export_sources()
```
It requests the source folder for the package `ref` (which is `pyreq`), but provides `short_paths` flag for the package `conanfile` which is `consumer`. Since `pyreq` does not have `shot_paths` flag, its sources were exported to the normal location, but `sources_folder` here points to a short path which even does not exist.

What is interesting, when I specify `short_paths` for `pyreq` too, it still does not work! When I create `pyreq` package, its `complete_recipe_sources` gets `conanfile.short_paths == False` and sources are still exported to their normal (non-short) location. Furthermore its "short" folder is created, but contains only `real_path.txt` which points to "long" folder. Thus when `complete_recipe_sources` is called from `consumer` context for `pyreq` it still can't find the sources directory.

**Comment by czoido** [comment]
Thanks a lot for reporting the bug @kdsx 
I have just made a PR that should fix the problem :)

**Comment by kdsx** [comment]
The fix in the PR helped me, thanks!

**Comment by memsharded** [comment]
Merged! Will be in Conan 1.20
